packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -460,6 +468,18 @@ class NavigatorState extends State<Navigator> {
     return _history.length > 1 || _history[0].willHandlePopInternally;
   }
 
+  void didStartUserGesture() { config.observer?.didStartUserGesture(); }
```

Comment:
Having said that, the general pattern of having paired calls to maintain state is pretty error prone, so maybe we should find a different approach.